tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -633,6 +633,7 @@ def batch_norm(inputs,
           renorm_clipping=renorm_clipping,
           renorm_momentum=renorm_decay,
           name=sc.name,
+          dtype=inputs.dtype.base_dtype,
```

Comment:
What do you think about this line?
I forgot to remove it in last commit. It adds possibility to use not-fp32-types with tf.contrib.slim.batch_norm and it also doesn't break backward compatibility. 
If you are working on batch norm and it's better to remove it from this pull request, please, add same line in your work.

Reply:
@reedwm Would you like to comment on this?

Reply:
We should not support fp16 and fp64 without tests. Also, for fused_batch_norm, the only fp16 case we support is where the input is fp16 and the scale and offset are in fp32. So if we support fp16 for non-fused batch norm, we should support that case.

In short, I'd remove the line. I plan on adding support for fp16 in layers shortly. If you want to do so, tell me so I don't repeat the work.